### PR TITLE
Convert tedium to Typescript

### PR DESCRIPTION
**Pull request by rictic**
I think there's a pretty decent case for converting our tools repos to typescript. The tools code is doing deterministic work with very structured data in a variety of ways, which is the kind of code where a type system can help out a lot.

I started with tedium because it's a fairly small and self contained repo, and it's not very community facing, so no one but us will be inconvenienced by the churn if we switch to TS and then back to JS if it doesn't work out.

I'd propose that we start with tedium and kick the tires some. Try to answer questions like:
- Is this code legible? 
- Does writing our tools code in typescript constrain us, or make it difficult for others to work with the code?
- How long does it take to get comfortable making changes to this sort of code? 
- Are the compiler warnings helpful or are we spending too much time just appeasing it?

If we like the answers to those questions we can consider migrating another tool to typescript. I've got a version of hydrolysis that's [been migrated here](https://github.com/rictic/hydrolysis/compare/master...typescript-experiment) if you're interested in seeing what that might look like.

A few notes:
- this compiles cleanly with no warnings or errors
- all configuration is in tsconfig.json, so to compile you just invoke `tsc` with no arguments, or `npm run build`, or even `npm run build:watch` to automatically rebuild as files change.
- the `atom-typescript` plugin is very awesome, I'd strongly recommend it. I believe that there's also  good plugins for vim and sublime text. On the mac there's [an Atom bug](https://github.com/atom/atom/issues/9663) that causes the plugin to be flaky. It's fixed in the Atom 1.3 beta.
- the typescript compiler doesn't do obfuscation, the output is just the ES6 that corresponds to this code

/cc @garlicnation @cdata @notwaldorf & anyone else who may be interested

### CODE REVIEW

**Comment by rictic** [comment]
Oh also @justinfagnani of course.

File: cleanup-passes/bower.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+import {ElementRepo} from '../element-repo.ts';
```
Comment (justinfagnani) [inline cleanup-passes/bower.ts]: I'd suggest sorting imports: external modules alphabetically, then relative imports, alphabetically.

File: tedium.ts [hunk 2]
```diff
@@ -25,20 +23,23 @@
...
+import * as GitHub from 'github-cache';
+import * as promisify from 'promisify-node';
+import * as nodegit from 'nodegit';
+import * as path from 'path';
+import * as ProgressBar from 'progress';
+import * as rimraf from 'rimraf';
+import {cleanup} from './cleanup';
+import {existsSync} from './cleanup-passes/util';
+import {ElementRepo} from './element-repo';
+import * as hydrolysis from 'hydrolysis';
```
Comment (justinfagnani) [inline tedium.ts]: sort imports

File: README.md [hunk 3]
```diff
@@ -6,7 +6,8 @@ A helpful bot for doing mass changes to Polymer team's github repos.
 
 ### Node v5:
 
-Tedium is written in ES2015, so you want the latest version of node:
+Tedium is written in Typescript and compiled to ES2015, so you want the
```
Comment (justinfagnani) [inline README.md:9]: mention installing tsc and tsd?

File: cleanup-passes/bower.ts (hunk 1 above)
Comment (rictic) [inline cleanup-passes/bower.ts]: Done, here and elsewhere.

File: README.md (hunk 3 above)
Comment (rictic) [inline README.md:9]: Covered by the `npm install && npm run init` below.

**Comment by rictic** [comment]
All comments addressed

**Pull request merged**